source/learn/simple-outlineview.html.markdown

```diff
@@ -0,0 +1,244 @@
+
+        }
+        return self;
+    }
+
+The [CPOutlineView API](http://www.cappuccino-project.org/learn/documentation/interface_c_p_outline_view.html) tells us clearly the required methods to implement and gives a good explanations as well:
+
+
+- `(int)outlineView:(CPOutlineView)outlineView numberOfChildrenOfItem:(id)item; ` 
+
+Returns the number of child items of a given item. If item is nil you should return the number of top level (root) items.
+
+
+- `(BOOL)outlineView:(CPOutlineView)outlineView isItemExpandable:(id)item;` 
+
+Returns YES if the item is expandable, otherwise NO.
+
+
+- `(id)outlineView:(CPOutlineView)outlineView child:(CPInteger)index ofItem:(id)item;` 
+
+Returns the child item at an index of a given item. If item is nil you should return the appropriate root item.
+
+
+- `(id)outlineView:(CPOutlineView)outlineView objectValueForTableColumn:(CPTableColumn)tableColumn byItem:(id)item;` 
+
+Returns the object value of the item in a given column.
+
+#### Implementing the methods
+Most probably the easiest method is the `numberOfChildrenOfItem` . 
+The `item` will be nil, if it is the root item. Otherwise, we simply use the size of the `CPMutableArray` called `_employees` .
+
+    :::objj
+    - (int)outlineView:(CPOutlineView)outlineView numberOfChildrenOfItem:(id)item
+    {
+        if (item == nil)
+        {
+            // We have one root item, the CEO only
+            return 1;
+        }
+        else
+        {
+            employee = item
+            return ([employee._employees count]);
+        }
+    }
+
+Next step to define whether the item has subitems. If it does, it will be expandable.
+
+    :::objj
+    - (BOOL)outlineView:(CPOutlineView)outlineView isItemExpandable:(id)item
+    {
+        if (item == nil)
+        {
+            return YES;
+        }
+        else
+        {
+            employee = item;
+            return ([employee._employees count] != 0);
+        }
+    }
+
+This method returns the child item at index of a item. If item is nil, returns the appropriate child item of the root object.
+
+    :::objj
+    - (id)outlineView:(CPOutlineView)outlineView child:(CPInteger)index ofItem:(id)item
+    {
+        if (item == nil)
+        {
+            // Our root item is the CEO
+            return ceo;
+        }
+        else
+        {
+            employee = item;
+            return (employee._employees[index]);
+        }
+    }
+
+Finally, we are assigning the object values to columns. Our first column is “NameColumn”, the second is “TitleColumn”. We will setup these names later in the Interface Builder.
+
+    :::objj
+    - (id)outlineView:(id)outlineView viewForTableColumn:(id)tableColumn item:(id)items
+    {
+        var identifier = [tableColumn identifier];
+
+        if (identifier == "NameColumn")
+        {
+            var view = [outlineView makeViewWithIdentifier:"NameCell" owner:self];
+            [[view textField] setStringValue:item._name ];
```

Comment:
If I define in the class `CPString _name @accessors` then I need to add `(property=name)`at the end otherwise I cannot refer to it as `name` only or did I miss something?